client/components/account-balances/balances-tab-panel.tsx

```diff
@@ -70,32 +76,52 @@ const AccountBalancesTabPanel: React.FC = () => {
 		);
 	}
 
-	return (
-		<TabPanel tabs={ depositCurrencyTabs }>
-			{ ( tab: BalanceTab ) => (
-				<Flex gap={ 0 } className="wcpay-account-balances__balances">
-					<BalanceBlock
-						type="available"
-						amount={ tab.availableFunds }
-						currencyCode={ tab.currencyCode }
-						isLoading={ isLoading }
-					/>
-					<BalanceBlock
-						type="pending"
-						amount={ tab.pendingFunds }
-						currencyCode={ tab.currencyCode }
-						isLoading={ isLoading }
-					/>
-					<BalanceBlock
-						type="reserved"
-						amount={ tab.reservedFunds }
-						currencyCode={ tab.currencyCode }
-						isLoading={ isLoading }
-					/>
-				</Flex>
-			) }
-		</TabPanel>
-	);
+	// Selected currency is not valid if it is not in the list of deposit currencies.
+	const isSelectedCurrencyValid =
+		selectedCurrency &&
+		depositCurrencyTabs.some( ( tab ) => tab.name === selectedCurrency );
+
+	// Wrap TabPanel to allow a full re-render when the selected currency changes.
+	// since TabPanel will not allow controlled tab selection, we use the initialTabName prop.
+	const TabPanelComp: React.FC = () => {
+		return (
+			<TabPanel
+				tabs={ depositCurrencyTabs }
+				onSelect={ onTabSelect }
+				initialTabName={
+					isSelectedCurrencyValid ? selectedCurrency : undefined
+				}
```

Comment:
If a string is passed to initialTabName that doesn't match with a tab name, no tab will be selected.